sec/insecure/insecure.go

```diff
@@ -5,64 +5,172 @@ package insecure
 
 import (
 	"context"
+	"fmt"
+	"github.com/libp2p/go-libp2p-core/network"
 	"net"
 
 	"github.com/libp2p/go-libp2p-core/peer"
 	"github.com/libp2p/go-libp2p-core/sec"
 
+	ggio "github.com/gogo/protobuf/io"
 	ci "github.com/libp2p/go-libp2p-core/crypto"
+	pb "github.com/libp2p/go-libp2p-core/sec/insecure/pb"
 )
 
 // ID is the multistream-select protocol ID that should be used when identifying
 // this security transport.
-const ID = "/plaintext/1.0.0"
+const ID = "/plaintext/2.0.0"
 
 // Transport is a no-op stream security transport. It provides no
-// security and simply mocks the security and identity methods to
-// return peer IDs known ahead of time.
+// security and simply mocks the security methods. Identity methods
+// return the local peer's ID and private key, and whatever the remote
+// peer presents as their ID and public key.
+// No authentication of the remote identity is performed.
 type Transport struct {
-	id peer.ID
+	id  peer.ID
+	key ci.PrivKey
 }
 
 // New constructs a new insecure transport.
-func New(id peer.ID) *Transport {
+func New(id peer.ID, key ci.PrivKey) *Transport {
```

Comment:
**This causes pretty severe API breakage.**

1. in the `Transport` option of the go-libp2p constructor.
2. elsewhere (e.g. go-tcp-transport) where this constructor is used directly.

Reply:
The private key is unnecessary here. The public key should do.

Reply:
We should deprecate the `New` function and make a `WithIdentity()` function that the public key, and returns a function of signature `func(id peer.ID) *Transport` that can then be used with the go-libp2p constructor.

Reply:
Adding a new constructor function and deprecating the old is definitely the way to go 👍

I added the private key to avoid returning `nil` from `LocalPrivateKey` - if we don't care about that then we can just take the public key.